src/Mapbender/WmsBundle/Component/WmsInstanceEntityHandler.php

```diff
@@ -79,17 +80,26 @@ public function setParameters(array $configuration = array())
             ->setSourceItem($layersourceroot)
             ->setSourceInstance($this->entity)
             ->setToggle(!isset($configuration["toggle"]) ? false : $configuration["toggle"])
-            ->setAllowtoggle(!isset($configuration["allowtoggle"]) ? true : $configuration["allowtoggle"]);
+            ->setAllowtoggle(!isset($configuration["allowtoggle"]) ? true : $configuration["allowtoggle"])
+            ->setPriority($num);
         $this->entity->addLayer($rootInstLayer);
-        foreach ($configuration["layers"] as $layerDef) {
+
+        foreach ($configuration['layers'] as $layerDef) {
+            $this->populateFromYaml($layerDef, $source, $layersourceroot, $rootInstLayer, $num);
+        }
+        return $this->entity;
+    }
+
+    public function populateFromYaml($configuration, WmsSource $source, WmsLayerSource $layersource, WmsInstanceLayer $instancelayer, $num)
+    {
             $num++;
-            $layersource = new WmsLayerSource();
-            $layersource->setSource($source)
-                ->setName($layerDef["name"])
-                ->setTitle($layerDef['title'])
-                ->setParent($layersourceroot)
-                ->setId($this->entity->getId() . '_' . $num);
-            if (isset($layerDef["legendurl"])) {
+            $childLayerSource = new WmsLayerSource();
+            $childLayerSource->setSource($source)
+                ->setName($configuration["name"])
+                ->setTitle($configuration['title'])
+                ->setParent($layersource)
+                ->setId($configuration['title'] . '_' . $num);
```

Comment:
This is not safe. Titles can repeat, and sometimes it makes perfect sense for them to repeat. E.g.  
`2018` => `Trees`  
`2018` => `Birds`  
`2019` => `Trees`  
`2019` => `Birds`  
(one might prefer using WMS dimensions here, but you get the point)

If we have to concatenate, I'd prefer concatenating the parent instance layer's id, which is readily available. Sth like
```php
      ->setId($layersource->getId() . '_' . $num)
```